app/src/main/java/com/android/unio/MainActivity.kt

```diff
@@ -66,7 +66,7 @@ fun UnioApp() {
         userRepositoryFirestore.getUserWithId(
             user.uid,
             {
-              if (it.hasProvidedAccountDetails) {
+              if (it.firstName.isNotEmpty()) {
```

Comment:
As @Aurelien9Code mentionned, checking that the `firstName` is not empty isn't entirely foolproof, or even semantically correct. Conceptually, the check should be on the existence of the document at `users/${user.uid}`. As mentionned in [the documentation](https://firebase.google.com/docs/firestore/query-data/get-data#kotlin+ktx_2) this is done inside the `.addOnSuccessListener` with a check to `document.exists()`. 

However, seeing the docs, I believe we should refactor all Firestore repositories to handle the case where the document does not exist. As it currently is implemented, all the success callbacks are called even when a document doesn't exist, but the non-existence of a document should be considered a failure.

Reply:
You're Right, checking for the existence of the first name is on top of being semantically incorrect, a very bad practise. We should work on that for the next Sprint. Typically a method in the User repo that checks the existence of the document will be a good idea as you explained.